diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -244,8 +245,7 @@ fun ASTNode.findAllNodesWithSpecificType(elementType: IElementType): List<ASTNod
  */
 fun ASTNode.isClassEnum(): Boolean = (psi as? KtClass)?.isEnum() ?: false
 
-fun ASTNode.isCompanionObject(): Boolean =
-        this.elementType == OBJECT_DECLARATION && this.hasChildOfType(OBJECT_KEYWORD) && this.firstChildNode.hasChildOfType(COMPANION_KEYWORD)
+fun ASTNode.isCompanionObject(): Boolean = this.psi is KtObjectDeclaration
```

Comment:
But it's not correct, this will return `true` for any object declaration. What you need is something like `(psi as? KtObjectDeclaration).isCompanion ?: false`